docs/grammar/SolidityLexer.g4

```diff
@@ -283,9 +284,7 @@ YulRParen: ')';
 YulAssign: ':=';
 YulPeriod: '.';
 YulComma: ',';
-// TODO: remove whitespace workaround once the parser disallows it.
-//@doc:name ->
-YulArrow: '->' | '-' YulWS+ '>';
+YulArrow: '->';
```

Comment:
Not sure if we can share tokens between the two lexer modes, hence kept it here with its own name.